tests/Unit/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/Test_TovStar.cpp

```diff
@@ -39,34 +42,46 @@ void verify_solution(const TovStar& solution, const std::array<double, 3>& x) {
   Mesh<3> mesh{brick.initial_extents()[0], Spectral::Basis::Legendre,
                Spectral::Quadrature::GaussLobatto};
   const auto domain = brick.create_domain();
-  verify_grmhd_solution(solution, domain.blocks()[0], mesh, 1.e-10, 1.234,
+  verify_grmhd_solution(solution, domain.blocks()[0], mesh, 1.e-7, 1.234,
                         1.e-4);
 }
 
 void test_tov_star(const TovCoordinates coord_system) {
+  Parallel::printf("Entered the function which tests");
   Parallel::register_classes_with_charm<
       RelativisticEuler::Solutions::TovStar>();
+  Parallel::register_classes_with_charm<
+      EquationsOfState::PolytropicFluid<true>>();
+  Parallel::printf("Registered with charm");
   const std::unique_ptr<evolution::initial_data::InitialData> option_solution =
       TestHelpers::test_option_tag_factory_creation<
           evolution::initial_data::OptionTags::InitialData,
           RelativisticEuler::Solutions::TovStar>(
           "TovStar:\n"
           "  CentralDensity: 1.0e-3\n"
-          "  PolytropicConstant: 100.0\n"
-          "  PolytropicExponent: 2.0\n"
+          "  EquationOfState:\n"
+          "    PolytropicFluid:\n"
+          "      PolytropicConstant: 100.0\n"
+          "      PolytropicExponent: 2.0\n"
           "  Coordinates: " +
           get_output(coord_system));
+  Parallel::printf("Got the thing from the option");
   const auto deserialized_option_solution =
       serialize_and_deserialize(option_solution);
+  Parallel::printf("Got the deserialized thing");
   const auto& solution =
       dynamic_cast<const RelativisticEuler::Solutions::TovStar&>(
           *deserialized_option_solution);
   {
     INFO("Semantics");
-    CHECK(solution == TovStar{1.0e-3, 100.0, 2.0, coord_system});
-    test_copy_semantics(solution);
-    TovStar star_copy = solution;
-    test_move_semantics(std::move(star_copy), solution);
+    CHECK(solution ==
+          TovStar{1.0e-3,
+                  std::make_unique<EquationsOfState::PolytropicFluid<true>>(
+                      100.0, 2.0),
+                  coord_system});
+    // test_copy_semantics(solution);
+    // TovStar star_copy = solution; // Broken
```

Comment:
could be restored by adding custom copy semantics that does a `get_clone` I wouldn't worry about it for now, just add a comment that `// Add support for arbitrary EOSes would require custom copy semantics that we haven't needed yet.`